spec/classes/plugin/dns_gandi_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'letsencrypt::plugin::dns_gandi' do
+  on_supported_os.each do |os, facts|
+    next if supported_os_gandi(facts[:os])
+
+    context "on #{os} based operating systems" do
+      let(:facts) { facts }
+      let(:params) { { 'api_key' => 'dummy-gandi-api-token' } }
+      let(:pre_condition) do
+        <<-PUPPET
+        class { 'letsencrypt':
+          email => 'foo@example.com',
+        }
+        PUPPET
+      end
+      let(:package_name) do
+        osname = facts[:os]['name']
+        osrelease = facts[:os]['release']['major']
+        osfull = "#{osname}-#{osrelease}"
+        case osfull
+        when 'Debian-11', 'Ubuntu-20.04'
+          'python3-certbot-dns-gandi'
+        end
+      end
+
+      context 'with required parameters' do
+        it do
+          if package_name.nil?
+            is_expected.not_to compile
+          else
+            is_expected.to compile.with_all_deps
+          end
+        end
+
+        describe 'with manage_package => true' do
+          let(:params) { super().merge(manage_package: true) }
+
+          it do
+            if package_name.nil?
+              is_expected.not_to compile
```

Comment:
A better way is to test for the error:
```suggestion
              is_expected.to compile.and_raise_error(/A Matcher For The Error/)
```